util/module_logger_test.go

```diff
@@ -0,0 +1,258 @@
+type dummyLogger struct {
+	msg        *logMsg
+	fieldKey   string
+	fieldValue interface{}
+	fields     log.LogFields
+}
+
+func (dl *dummyLogger) Debug(args ...interface{}) {
+	dl.msg = &logMsg{args: args, level: DebugLevel}
+}
+
+func (dl *dummyLogger) Debugf(format string, args ...interface{}) {
+	dl.msg = &logMsg{format: format, args: args, level: DebugLevel}
+}
+
+func (dl *dummyLogger) Info(args ...interface{}) {
+	dl.msg = &logMsg{args: args, level: InfoLevel}
+}
+
+func (dl *dummyLogger) Infof(format string, args ...interface{}) {
+	dl.msg = &logMsg{format: format, args: args, level: InfoLevel}
+}
+
+func (dl *dummyLogger) Warn(args ...interface{}) {
+	dl.msg = &logMsg{args: args, level: WarnLevel}
+}
+
+func (dl *dummyLogger) Warnf(format string, args ...interface{}) {
+	dl.msg = &logMsg{format: format, args: args, level: WarnLevel}
+}
+
+func (dl *dummyLogger) Error(args ...interface{}) {
+	dl.msg = &logMsg{args: args, level: ErrorLevel}
+}
+
+func (dl *dummyLogger) Errorf(format string, args ...interface{}) {
+	dl.msg = &logMsg{format: format, args: args, level: ErrorLevel}
+}
+
+func (dl *dummyLogger) Fatal(args ...interface{}) {
+	dl.msg = &logMsg{args: args, level: FatalLevel}
+}
+
+func (dl *dummyLogger) Fatalf(format string, args ...interface{}) {
+	dl.msg = &logMsg{format: format, args: args, level: FatalLevel}
+}
+
+func (dl *dummyLogger) Panic(args ...interface{}) {
+	dl.msg = &logMsg{args: args, level: PanicLevel}
+}
+
+func (dl *dummyLogger) Panicf(format string, args ...interface{}) {
+	dl.msg = &logMsg{format: format, args: args, level: PanicLevel}
+}
+
+func (dl *dummyLogger) WithField(key string, value interface{}) log.Logger {
+	dl.fieldKey = key
+	dl.fieldValue = value
+	return dl
+}
+
+func (dl *dummyLogger) WithFields(keyValues log.LogFields) log.Logger {
+	dl.fields = keyValues
+	return dl
+}
+
+func (dl *dummyLogger) Fields() log.Fields {
+	return nil
+}
+
+func (dl *dummyLogger) assertMsg(t assert.TestingT, msgLevel Level, args ...interface{}) {
+	assert.Equal(t, dl.msg.level, msgLevel, "Message emitted with wrong level.")
+	assert.Equal(t, dl.msg.args, args, "Message emitted with wrong args.")
+}
+
+func (dl *dummyLogger) assertMsgf(t assert.TestingT, msgLevel Level, format string, args ...interface{}) {
+	dl.assertMsg(t, msgLevel, args...)
+	assert.Equal(t, dl.msg.format, format, "Message emitted with wrong format.")
+}
+
+func (dl *dummyLogger) assertNoMsg(t assert.TestingT) {
+	assert.Nil(t, dl.msg, "Unexpected message emitted.")
+}
+
+type ModuleTestSuite struct {
+	suite.Suite
+	dl *dummyLogger
+	ml *moduleLogger
+}
+
+func (suite *ModuleTestSuite) SetupTest() {
+	suite.dl = &dummyLogger{}
+	suite.ml = NewModuleLogger(suite.dl)
+	suite.ml.SetModuleLevel("testdebug", DebugLevel)
+	suite.ml.SetModuleLevel("testpanic", PanicLevel)
+	suite.ml.SetModuleLevel("testoff", OffLevel)
+}
+
+func (suite *ModuleTestSuite) assertMsg(msgLevel Level, args ...interface{}) {
+	suite.dl.assertMsg(suite.T(), msgLevel, args...)
```

Comment:
Yeah, I saw when writing the tests for options that there is a Logger mock already; I'll rewrite this to use that instead.